src/shared/extensions/git.ts

```diff
@@ -138,14 +131,30 @@ export class GitExtension {
     // TODO: use `ChildProcess` to execute git
     public executeCommand(): void {}
 
+    private async validateApi(message: string): Promise<GitTypes.API | undefined>
+    private async validateApi(message: Error): Promise<GitTypes.API | never>
+    private async validateApi(message: string | Error): Promise<GitTypes.API | undefined | never> {
```

Comment:
Do we really benefit from 3 constructors for a private function in the class? The last one looks like it'll do the job just fine.

Looking at impls later, I feel like this could just return a boolean. Why have the `_api` tied to the class if you're not going to use it elsewhere?